File: android/src/main/java/com/swmansion/gesturehandler/OnTouchEventListener.java
package com.swmansion.gesturehandler;

import android.view.MotionEvent;

import com.swmansion.gesturehandler.GestureHandler;

public interface OnTouchEventListener<T extends GestureHandler> {
  void onTouchEvent(T handler, MotionEvent event);
  void onStateChange(T handler, int newState, int oldState);
}

File: android/lib/src/main/java/com/swmansion/gesturehandler/OnTouchEventListener.kt
package com.swmansion.gesturehandler

import android.view.MotionEvent

interface OnTouchEventListener {
  fun <T : GestureHandler<T>> onTouchEvent(handler: T, event: MotionEvent)
  fun <T : GestureHandler<T>> onStateChange(handler: T, newState: Int, oldState: Int)
  fun <T : GestureHandler<T>> onPointerEvent(handler: T)
}
